### PR TITLE
Update docstrings in integration plugin to clarify restriction

### PR DESCRIPTION
**Pull request by gavinevans**
Description
Update the docstrings in the integration plugin to clarify that it only integrates positive values.

Testing:
 - [x] Ran tests and they passed OK
 - [x] Added new tests for the new feature(s)

### CODE REVIEW

File: lib/improver/tests/utilities/test_mathematical_operations.py [hunk 1]
```diff
@@ -333,7 +384,7 @@ def test_start_point_positive(self):
             result.coord("height").points, np.array([20.]))
         self.assertArrayAlmostEqual(result.data, expected)
 
-    def test_start_point_negative(self):
+    def test_start_point_negative_direction(self):
```
Comment (bayliffe) [inline lib/improver/tests/utilities/test_mathematical_operations.py]: This doc-string and the one above appear to be in conflict (as do all the positive/negative test doc-strings below). Both say that integrating downwards in a height coordinate is equivalent to their direction (both positive and negative). I think they are both valid as the cubes have been ordered differently, but the statement reads as if we are making a general point about integration in a height coordinate. Could we just tweak them to something like:
```
(equivalent to integrating downwards for the height coordinate in the input cube)
```
which makes it sound more specific to this test case?

File: lib/improver/tests/utilities/test_mathematical_operations.py [hunk 2]
```diff
@@ -359,7 +410,7 @@ def test_start_point_negative(self):
             result.coord("height").points, np.array([5.]))
         self.assertArrayAlmostEqual(result.data, expected)
 
-    def test_end_point_positive(self):
+    def test_end_point_positive_direction(self):
```
Comment (bayliffe) [inline lib/improver/tests/utilities/test_mathematical_operations.py]: This doc-string says negative direction, but the title says positive (it appears that it is positive, so the doc-string needs fixing).

File: lib/improver/utilities/mathematical_operations.py [hunk 3]
```diff
@@ -38,7 +38,12 @@
 
 
 class Integration(object):
-    """Perform integration along a chosen coordinate."""
+    """Perform integration along a chosen coordinate. This class currently
+    only supports the integration of positive values only, in order to
```
Comment (bayliffe) [inline lib/improver/utilities/mathematical_operations.py]: Too many **onlys**.

File: lib/improver/tests/utilities/test_mathematical_operations.py (hunk 1 above)
Comment (gavinevans) [reply lib/improver/tests/utilities/test_mathematical_operations.py]: I've amended this docstring and the docstrings for the other tests as suggested.

File: lib/improver/tests/utilities/test_mathematical_operations.py (hunk 2 above)
Comment (gavinevans) [reply lib/improver/tests/utilities/test_mathematical_operations.py]: Amended docstring.

File: lib/improver/utilities/mathematical_operations.py (hunk 3 above)
Comment (gavinevans) [reply lib/improver/utilities/mathematical_operations.py]: Removed an `only`.

**Pull request merged**